pkg/sdkstats/sdkstats.go

```diff
@@ -2,161 +2,120 @@ package sdkstats
+}
 
-			TotalCommits, err := strconv.Atoi(repoMap["TotalCommits"])
-			Stars, err := strconv.Atoi(repoMap["Stars"])
-			if err != nil {
-				return nil, err
-			}
+//GetStats returns []RepoMetaData populated with Code results from Github Search Code API, for a given search string
+func GetStats(client *github.Client, rq RepoMetadataQuery) ([]RepoMetadata, error) {
+	var repoList []RepoMetadata
+	var sdkVersion string
+	var skipVersion int
 
-			repoDetails := &RepoMetadata{
-				FullName:     fullName,
-				ProjectType:  projectType,
-				SDKVersion:   sdkVersion,
-				Stars:        Stars,
-				CreatedAt:    repoMap["CreatedAt"],
-				PushedAt:     repoMap["PushedAt"],
-				TotalCommits: TotalCommits,
+	for _, q := range rq.Queries {
+		searchOp, err := Search(client, q)
+		if err != nil {
+			return nil, err
+		}
+		if strings.Contains(q, "Gopkg.toml") {
+			skipVersion = 1
+		}
+		for j := 0; j < len(searchOp); j++ {
+			for i := 0; i < len(searchOp[j].CodeResults); i++ {
+				if skipVersion == 1 {
+					sdkVersion = "N/A"
+				} else {
+					sdkVersion, err = rq.VersionParser.ParseVersion(searchOp[j].CodeResults[i], rq.ProjectType)
+					if err != nil {
+						return nil, err
+					}
+				}
+				repoDetails := &RepoMetadata{
+					URL:         searchOp[j].CodeResults[i].GetRepository().GetURL(),
+					Name:        searchOp[j].CodeResults[i].GetRepository().GetName(),
+					Owner:       searchOp[j].CodeResults[i].GetRepository().GetOwner().GetLogin(),
+					ProjectType: rq.ProjectType,
+					SDKVersion:  sdkVersion,
+				}
+				repoDetails, err = GetRepoDetails(client, repoDetails)
+				if err != nil {
+					return nil, err
+				}
+				repoList = append(repoList, *repoDetails)
 			}
-			repoList = append(repoList, *repoDetails)
-
 		}
 	}
-
 	return repoList, nil
 }
 
 //Search returns Github API results for Code Search, for a given Query string
 func Search(client *github.Client, q string) ([]*github.CodeSearchResult, error) {
-
-	var searchOp []*github.CodeSearchResult
+	var searchop []*github.CodeSearchResult
 	ctx := context.Background()
-
 	opts := &github.SearchOptions{TextMatch: true,
 		ListOptions: github.ListOptions{
 			PerPage: 100,
 		},
 	}
+	//Below Logic enables to collate Search results from all pages returned from API call.
 	for {
 		op, resp, err := client.Search.Code(ctx, q, opts)
 		if err != nil {
 			return nil, err
 		}
-		searchOp = append(searchOp, op)
+		searchop = append(searchop, op)
 		if resp.NextPage == 0 {
 			break
 		}
 		opts.Page = resp.NextPage
 	}
-	return searchOp, nil
-
+	return searchop, nil
 }
 
-//GetRepoDetails Returns map[string][string] with Stars,CreatedAt,PushedAt, and TotalCommits details for a given onwer, and repo
-func GetRepoDetails(client *github.Client, owner string, name string) map[string]string {
-
-	repoMap := map[string]string{}
-
+//GetRepoDetails Returns []RepoMetaData with Stars,CreatedAt,PushedAt, and TotalCommits details for a given owner, and repo
```

Comment:
A couple of nits, not super important, but good to instill good habits (this is as much a reminder for me as is it for anyone else reading this :slightly_smiling_face:)
1. Include a space after the comment marker `// `
2. The first word of a comment that describes an identifier should be the identifier, so good to go there :+1: 
3. Use full sentences that end with a period
4. It's documentation for humans, so write what seems most natural and helpful.

With that in mind, perhaps this:
```suggestion
// GetRepoDetails uses client to look up repository metadata about repo.
// The returned RepoMetadata struct includes details such as the
// repository's stars, timestamps when it was created and most recently
// pushed to, and the total number of commits to the repo. If an error
// occurs, *RepoMetadata will be nil, and the error will be non-nil.
```

Reply:
Sounds reasonable, will include.

Reply:
Another point is that it is most important to pay attention to the doc conventions on exported functions. Since that's what users of the library will see.

Comments on unexported functions are still nice, but are only helpful to other developers and maintainers of the library.